slowquant/qiskit_interface/custom_ansatz.py

```diff
@@ -143,16 +154,40 @@ def fUCC(
         ansatz_options: Ansatz options.
 
     Returns:
-        Factorized UCCSD ansatz circuit and R parameters needed for gradients.
+        Factorized UCC ansatz circuit and R parameters needed for gradients.
     """
-    valid_options = ("n_layers",)
+    valid_options = ("n_layers", "S", "D", "SAGS", "pD", "GpD", "SAS")
     for option in ansatz_options:
         if option not in valid_options:
             raise ValueError(f"Got unknown option for fUCC, {option}. Valid options are: {valid_options}")
-    if not isinstance(mapper, JordanWignerMapper):
-        raise ValueError(f"fUCC only implemented for JW mapper, got: {type(mapper)}")
     if "n_layers" not in ansatz_options.keys():
         raise ValueError("fUCC require the option 'n_layers'")
+    do_S = False
+    do_SAS = False
+    do_SAGS = False
+    do_D = False
+    do_pD = False
+    do_GpD = False
+    if "S" in ansatz_options.keys():
+        if ansatz_options["S"]:
+            do_S = True
+    if "SAS" in ansatz_options.keys():
```

Comment:
Does it makes sense to allow all kind of singles? Currently, you could have S and SAS in one circuit

Reply:
It doesnt really make sense, but also only something that can be reached if someone asks for it very explicitly.